### PR TITLE
Changing Notes CSON to Markdown

### PR DESCRIPTION
**Pull request by maykefreitas**
Changing notes files to use .md instead of .cson. https://github.com/BoostIO/Boostnote/issues/490

Type of changes
[ ] Bug fix (Change that fixed an issue)
[x] Breaking change (Change that can cause existing functionality to change)
[ ] Improvement (Change that improves the code. Maybe performance or development improvement)
[ ] Feature (Change that adds new functionality)
[ ] Documentation change (Change that modifies documentation. Maybe typo fixes)
Checklist:
[x] My code follows the project code style
[x] I have written test for my code and it has been tested
[x] All existing tests have been passed
[ ] I have attached a screenshot/video to visualize my change if possible

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#490: Store notes as .md files instead of .cson files](https://issuehunt.io/repos/53266139/issues/490)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by Flexo013** [comment]
I will also put @arcturus140's question here:
> what happens with existing .cson notes?

Additionally could add this checklist to your PR description?

```
Type of changes
[ ] Bug fix (Change that fixed an issue)
[ ] Breaking change (Change that can cause existing functionality to change)
[ ] Improvement (Change that improves the code. Maybe performance or development improvement)
[ ] Feature (Change that adds new functionality)
[ ] Documentation change (Change that modifies documentation. Maybe typo fixes)
Checklist:
[ ] My code follows the project code style
[ ] I have written test for my code and it has been tested
[ ] All existing tests have been passed
[ ] I have attached a screenshot/video to visualize my change if possible
```

**Comment by arcturus140** [comment]
Since this format is incompatible with the existing one it requires to increase the major version for proper handling.

**Comment by ZeroX-DG** [comment]
Well it's better to have a tool for converting cson to markdown (separated from the main app). The main app will only parse any markdown file and show a small list of cson file not being parsed and saying users need to use a tool to convert them. What do you guys think?

**Comment by arcturus140** [comment]
@ZeroX-DG we don't even need to list files, jsonData.version already tells which format we have. Then show a pop up to force an upgrade of the entire storage location. 

if external tool or internal is only a matter of convenience.

**Comment by arcturus140** [comment]
the thing is, once upgraded there's no turning back, so we force everyone to update to the latest version - shared devices and team members...

I'm sticking with an older version, because of some bugs in the later. I'm sure I'm not the only one, since making updates optional is also a highly wanted feature. 

but we gotta do it anyway at some point, so...

**Comment by Hadrien-lcrx** [comment]
@ZeroX-DG Someone seems to have worked on Boostnote `cson` to `md` conversion. It seems improvable but may be useful: https://gist.github.com/weaming/32b7b62956b304ff66fbae396ca9c86b

**Comment by eromoe** [comment]
if boostnote use md as default format, so that I can directly map path to jekyll , make blog easily .
And this https://github.com/BoostIO/Boostnote/issues/3117 would be easy too.

File: browser/main/lib/dataApi/noteIO.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const _ = require('lodash')
+const fs = require('fs-plus')
+const CSON = require('@rokt33r/season')
+
+function createMarkdown (input) {
+  const content = input.content || ''
+  const meta = _.omit(input, ['key', 'storage', 'content'])
+  const result = CSON.stringify(meta)
+
+  return `<!--\n${result}\n-->${content ? `\n\n${content}` : ''}`
```
Comment (c33s) [inline browser/main/lib/dataApi/noteIO.js:10]: why not using the frontmattre format or at least make it configurable?

**Comment by c33s** [comment]
any plans to merge and release this?

**Pull request closed without merging**